### PR TITLE
Prune sig-cluster-lifecycle github teams

### PR DESCRIPTION
**Pull request by timothysc**
Less is more, prune the lists! 

/assign @roberthbailey @kris-nova

### CODE REVIEW

File: sigs.yaml [hunk 1]
```diff
@@ -678,20 +678,10 @@ sigs:
...
-        description: API Changes and Reviews
-      - name: sig-cluster-lifecycle-bugs
-        description: Bug Triage and Troubleshooting
-      - name: sig-cluster-lifecycle-feature-requests
-        description: Feature Requests
-      - name: sig-cluster-lifecycle-misc
-        description: General Discussion
+      - name: sig-cluster-lifecycle
+        description: Notify group
       - name: sig-cluster-lifecycle-pr-reviews
```
Comment (roberthbailey) [inline sigs.yaml]: Aren't we going to delete the pr-reviews one too?

File: sigs.yaml [hunk 2]
```diff
@@ -678,20 +678,10 @@ sigs:
...
-      - name: sig-cluster-lifecycle-api-reviews
-        description: API Changes and Reviews
-      - name: sig-cluster-lifecycle-bugs
-        description: Bug Triage and Troubleshooting
-      - name: sig-cluster-lifecycle-feature-requests
-        description: Feature Requests
-      - name: sig-cluster-lifecycle-misc
-        description: General Discussion
+      - name: sig-cluster-lifecycle
+        description: Notify group
```
Comment (roberthbailey) [inline sigs.yaml]: Are we supposed to keep this one around? Or just nuke all of them?

File: sigs.yaml (hunk 1 above)
Comment (timothysc) [reply sigs.yaml]: I'm following the pattern that we might want to separate reviews from global notifications.

File: sigs.yaml (hunk 2 above)
Comment (timothysc) [reply sigs.yaml]: Yes we need at least one.

**Comment by cblecker** [comment]
/approve
/hold

Feel free to remove the hold when ready

**Comment by timothysc** [comment]
@cblecker this is blocked by https://github.com/kubernetes/org/issues/25

**Comment by cblecker** [comment]
@timothysc you are fixed up and should be good to go perms wise

**Comment by timothysc** [comment]
all done, go-for merge

**Pull request merged**